src/dodal/devices/areadetector/adaravis.py

```diff
@@ -97,3 +115,109 @@ def _prime_hdf(self) -> None:
         # Revert settings to previous values
         for signal, value in reversed(reset_to.items()):
             signal.set(value).wait(timeout=10)
+
+
+class TriggerSourceMako(str, Enum):
+    freerun = "Freerun"
+    line_1 = "Line1"
+    line_2 = "Line2"
+    fixed_rate = "FixedRate"
+    software = "Software"
+    # action_0 = "Action0"
+    # action_1 = "Action1"
```

Comment:
What do we want to do about this commented-out code? Remove or uncomment?

Reply:
@coretl think this may be one for you- looks like the initial commit had line_2 commented out, then it was restored and action_0, action_1 were commented out.

If [this is the relevant Mako driver](https://cdn.alliedvision.com/fileadmin/content/documents/products/cameras/various/features/Mako_U_Features_Reference_V1.1.0.pdf), it doesn't have action, but the [GigE camera has this and more](https://cdn.alliedvision.com/fileadmin/content/documents/products/cameras/various/features/GigE_Features_Reference.pdf).

Reply:
The problem is that this is camera firmware dependent. I think we need to find a way to make Enum strings be non-exhaustive, maybe in the signal backend converter we subclass the supplied Enum to add extra entries if it they exist? Then this code can just be `fixed_rate` and `line_1` and the others can be added at runtime as we don't care what they are

Reply:
Once this is merged, we should be able to reduce to just the values that are being used.
https://github.com/bluesky/ophyd-async/pull/147

Getting the signal may return values outside of the enum, but putting the limited enum should be ok?

Reply:
> Getting the signal may return values outside of the enum

I guess the signal should probably be type-hinted as `str` then - or at least something that gives an indication this isn't a standard enum... Otherwise this is bound to trip people up eventually.